src/main/java/ru/bclib/util/TagHelper.java

```diff
@@ -58,11 +58,11 @@ public static void addTags(Block block, Tag.Named<Block>... tags) {
 		return builder;
 	}
 	
-	public static Map<ResourceLocation, Tag.Builder> apply(String entry, Map<ResourceLocation, Tag.Builder> tagsMap) {
+	public static Map<ResourceLocation, Tag.Builder> apply(String directory, Map<ResourceLocation, Tag.Builder> tagsMap) {
 		Map<ResourceLocation, Set<ResourceLocation>> endTags = null;
-		if (entry.equals("block")) {
+		if ("tags/blocks".equals(directory)) {

```

Comment:
Why "block" was changed to "tags/blocks" (the same for items)? Is this internal Mojang change? And why string is compared with variable instead of variable compared with string like in original code?

Reply:
The `name`-Attribute in `TagLoader` was removed, what remains is the `directory` attribute (and a ref to the registry itself). The `directory` appears to separate blocks/items based on those new strings. Maybe a better solution is to check for both?

Reply:
This actually requires testing, probably with simple System.out.println(directory); If this will return "tags/blocks" - the code will work correctly

Reply:
I agree. The change is based on `net.minecraft.tags.ItemTags`. The code creates the `Registry.ITEM_REGISTRY` using **tags/items** as the directory name (created using `StaticTags.create` which instantiates a new `StaticTagHelper` using that directory name). Same goes for `net.minecraft.tags.BlockTags`. The `net.minecraft.tags.TagManager` class will use those instances to call `TagLoader.load`.

Reply:
The following values for `directory` are logged when started with BetterEnd:
```
[16:36:23] [Worker-Main-14/INFO] (Logger) [bclib] Directory: tags/fluids
[16:36:23] [Worker-Main-10/INFO] (Logger) [bclib] Directory: tags/game_events
[16:36:23] [Worker-Main-9/INFO] (Logger) [bclib] Directory: tags/entity_types
[16:36:23] [Worker-Main-8/INFO] (Logger) [bclib] Directory: tags/items
[16:36:23] [Worker-Main-13/INFO] (Logger) [bclib] Directory: tags/blocks
[16:36:27] [Worker-Main-8/INFO] (Logger) [bclib] Directory: tags/game_events
[16:36:27] [Worker-Main-11/INFO] (Logger) [bclib] Directory: tags/fluids
[16:36:27] [Worker-Main-13/INFO] (Logger) [bclib] Directory: tags/entity_types
[16:36:27] [Worker-Main-15/INFO] (Logger) [bclib] Directory: tags/items
[16:36:27] [Worker-Main-14/INFO] (Logger) [bclib] Directory: tags/blocks
```

Reply:
Hmm. Ok. It need to change equals then.

Reply:
The string-literal can never be `null` so even if `directory` would ever happen to be `null` (which is probably never the case) this code would not crash. Sorry, it is a force of habit. If it is against your style guides I'll swap it back :)